cafebabel/static/js/script.js

```diff
@@ -26,6 +26,50 @@ Array.from(document.querySelectorAll('a')).forEach(a => {
     a.setAttribute('target', '_blank')
 })
 
+/* smooth scroll to anchor https://stackoverflow.com/a/17733311/6481285 */
+function animate(elem, style, unit, from, to, time, prop) {
+  if (!elem) {
+    return
+  }
+  var start = new Date().getTime(),
+    timer = setInterval(() => {
+      var step = Math.min(1, (new Date().getTime() - start) / time)
+      if (prop) {
+        elem[style] = from + step * (to - from) + unit
+      } else {
+        elem.style[style] = from + step * (to - from) + unit
+      }
+      if (step === 1) {
+        clearInterval(timer)
+      }
+    }, 25)
```

Comment:
Document magic numbers.

Reply:
you mean we don't know what `25` refers to? I don't :-(